src/box/memtx_engine.h

```diff
@@ -110,10 +110,16 @@ struct memtx_engine {
 	struct checkpoint *checkpoint;
 	/** The directory where to store snapshots. */
 	struct xdir snap_dir;
+	/** The directory where to store the sort data. */
+	struct xdir sortdata_dir;
 	/** Limit disk usage of checkpointing (bytes per second). */
 	uint64_t snap_io_rate_limit;
 	/** Skip invalid snapshot records if this flag is set. */
 	bool force_recovery;
+	/** Save and load the sort data. */
+	bool sort_data_enabled;
+	/** The memtx index sort data reader (only non-NULL on recovery). */
+	struct memtx_sort_data_reader *msdr;
```

Comment:
Sort data reader shouldn't be stored in the engine because it's used only on recovery. Also, non-standard abbreviations like "msdr" and "msd" look bewildering, better avoid them.

I see no reason in `sortdata_dir` because a sort data file should always have the same signature as the corresponding snapshot file; they should be created and deleted together.

Reply:
> The sort data reader is used from the engine struct during PK recovery in order to fill the old2new tuple address map. But it can be made a separated engine-specific (or generic) `build_next` argument, of course.

Makes sense. Let's leave it in the engine then.

> 
> Separated dirs are made for independent garbage collection (see the [comment above](https://github.com/tarantool/tarantool/pull/11133#discussion_r2041292840)).

I'm not convinced we need to implement independent garbage collection for `.snap` and `.sortdata` files because a `.sortdata` file should exist only if there's a corresponding `.snap` file.

Reply:
Regarding the garbage collection, see the [comment above](https://github.com/tarantool/tarantool/pull/11133#discussion_r2052049629).